ckan/views/group.py

```diff
@@ -0,0 +1,1160 @@
+    '''This is an interface to manipulate data from the database
+    into a format suitable for the form (optional)'''
+    return lookup_group_plugin(group_type).db_to_form_schema()
+
+
+def _setup_template_variables(context, data_dict, group_type=None):
+    return lookup_group_plugin(group_type).\
+        setup_template_variables(context, data_dict)
+
+
+def _new_template(group_type):
+    return lookup_group_plugin(group_type).new_template()
+
+
+def _about_template(group_type):
+    return lookup_group_plugin(group_type).about_template()
+
+
+def _read_template(group_type):
+    return lookup_group_plugin(group_type).read_template()
+
+
+def _history_template(group_type):
+    return lookup_group_plugin(group_type).history_template()
+
+
+def _edit_template(group_type):
+    return lookup_group_plugin(group_type).edit_template()
+
+
+def _activity_template(group_type):
+    return lookup_group_plugin(group_type).activity_template()
+
+
+def _admins_template(group_type):
+    return lookup_group_plugin(group_type).admins_template()
+
+
+def _bulk_process_template(group_type):
+    return lookup_group_plugin(group_type).bulk_process_template()
+# end hooks
+
+
+def _replace_group_org(string):
+    ''' substitute organization for group if this is an org'''
+    return re.sub('^group',
+                  lookup_group_controller(_guess_group_type()), string)
+
+
+def _action(action_name):
+    ''' select the correct group/org action '''
+    return get_action(_replace_group_org(action_name))
+
+
+def _check_access(action_name, *args, **kw):
+    ''' select the correct group/org check_access '''
+    return check_access(_replace_group_org(action_name), *args, **kw)
+
+
+def _render_template(template_name, group_type):
+    ''' render the correct group/org template '''
+    return base.render(
+        _replace_group_org(template_name),
+        extra_vars={'group_type': group_type})
+
+
+def _force_reindex(grp):
+    ''' When the group name has changed, we need to force a reindex
+    of the datasets within the group, otherwise they will stop
+    appearing on the read page for the group (as they're connected via
+    the group name)'''
+    group = model.Group.get(grp['name'])
+    for dataset in group.packages():
+        search.rebuild(dataset.name)
+
+
+def _guess_group_type(expecting_name=False):
+    """
+            Guess the type of group from the URL.
+            * The default url '/group/xyz' returns None
+            * group_type is unicode
+            * this handles the case where there is a prefix on the URL
+              (such as /data/organization)
+        """
+    parts = [x for x in request.path.split('/') if x]
+
+    idx = 0
+    if expecting_name:
+        idx = -1
+
+    gt = parts[idx]
+
+    return gt
+
+
+def add_group_type(group_type):
+    group_types.append(group_type)
+
+
+def index():
+    group_type = _guess_group_type()
+    page = h.get_page_number(request.params) or 1
+    items_per_page = 21
```

Comment:
Maybe it's time to use `int(config.get('ckan.datasets_per_page', 20))`
PS don't you know, why we are using 21 here?

Reply:
dont know why is 21, but i will change it as your suggestion to 20

Reply:
I suppose, that it related to masonry grid, which we used previously. There were 3 blocks in a row, so 21 groups are required visually. I think we've got rid of masonry for bs3

Reply:
yes, i notice that tests are written in that manner. @gocemitevski create PR for updating masonry, after the merge, i will fix the tests

Reply:
@tino097 done. https://github.com/ckan/ckan/pull/4114